lib/BasicAnalyses/CSVAliasAnalysis.cpp

```diff
@@ -0,0 +1,109 @@
+/// \file CSVAliasAnalysis.cpp
+/// \brief Decorate memory accesses with information about CSV aliasing.
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/MDBuilder.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/IR/Value.h"
+
+#include "revng/BasicAnalyses/CSVAliasAnalysis.h"
+#include "revng/Support/Assert.h"
+
+using namespace llvm;
+using CSVAAP = CSVAliasAnalysisPass;
+
+PreservedAnalyses CSVAAP::run(Function &F, FunctionAnalysisManager &FAM) {
```

Comment:
The new version seems much nicer. Is it just refactoring or do you think it might have changes in semantics worth reviewing?

Reply:
It was definitely worth reviewing. Not only did we have chase some subtle bugs, but the design dramatically improved and a better separation of concern has been achieved. Together with `SegregateDirectStackAccessesPass`, these two passes now look much cleaner.